[Gareth Ong Ting Yi] iP

GarethOng opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER*  FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :drooling_face:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon) 


If you Java programmer, you can use it to practice Java too. Here's the  `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Greeting and Bye function added

Commit:
add task function & list function

Commit:
mark and unmark function

Commit:
ToDos, Events, Deadlines

Commit:
exception handling

Commit:
delete function

Commit:
implement automated UI testing

Commit:
save function

Commit:
date parser implemented

Commit:
Merge branch 'branch-Level-8'

Commit:
date function

Commit:
add more OOP

Commit:
package commands

Commit:
modify update

Commit:
Merge branch 'add-gradle-support' of github.com:nus-cs2103-AY2223S1/ip

Commit:
add gradle support

Commit:
added JUnit test

Commit:
Add JavaDoc

Commit:
Adjust formatting to fit CodingStandard

Commit:
Add find feature

Commit:
Merge branch 'Level-9'

Commit:
add CheckStyle

Commit:
add GUI

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Merge branch 'branch-Level-10'

aarontangr commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,39 @@
+     *         "[T] [status] task"
+     */
+    @Override
+    public String toString() {
+        return "[T] "
+                + this.getStatusIcon()
+                + " "
+                + super.description;
+    }
```
Can use `String.format`

aarontangr commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,39 @@
+     */
+    @Override
+    public String toSaveVersion() {
+        if (this.isDone()) {
+            return "T|1|"
+                    + super.description
+                    + "\n";
+        } else {
+            return "T|0|"
+                    + super.description
+                    + "\n";
+        }
```
Can use `String.format`

aarontangr reviewed (commented):
Looks good.
Some cosmetic changes requested.

ryanczx commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+        Object[] taskArr = this.taskList.toArray();
+        String result = "";
+        for (int i = 0; i < this.taskList.size(); i++) {
+            result += (i + 1)
```
Nice that you used brackets here to show grouping to make the code more obvious!

ryanczx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,89 @@
+    public void load(TaskList taskList) throws FileNotFoundException, ImproperFormatException {
+        Scanner reader = new Scanner(this.storageFile);
+        while (reader.hasNext()) {
+            String curr = reader.nextLine();
+            String[] currArr = curr.split("[|]");
+            String isDone = currArr[1];
+            String currKeyword = currArr[0];
```
Great that you organized your code this way to simplify it!

ryanczx reviewed (commented):
Looks good overall!

Commit:
Add assertion

Add assertion to Storage, TaskList and Ui to document important assumptions.

Commit:
Improve code quality

Improve code quality in Parser and Storage function

Commit:
Add streams

Utilise Java streams API for finTask and generateSave. Wrote test cases for both methods to ensure no errors.

Commit:
Add sort

Add sort function that allows users to sort either their deadlines or events in either a chronological or lexigraphical manner in either increasing or decreasing manner

GarethOng committed:
Merge pull request #1 from GarethOng/branch-A-Assertions

Add assertion

GarethOng committed:
Merge branch 'master' into branch-A-CodeQuality

GarethOng committed:
Merge pull request #2 from GarethOng/branch-A-CodeQuality

Improve code quality

GarethOng committed:
Merge branch 'master' into branch-A-Streams

GarethOng committed:
Merge pull request #3 from GarethOng/branch-A-Streams

Add streams

GarethOng committed:
Merge branch 'master' into branch-C-Sort

GarethOng committed:
Merge pull request #4 from GarethOng/branch-C-Sort

Add sort

Commit:
Handle merge conflicts

Current conflicts in implementation of the Parser function. Due to the addition of new keyword "sort" conflicts with the Parser version in branch-A-CodeQuality.

Add "sort" keyword following the improved code quality inside branch-A-CodeQuality.

Commit:
CheckStyle

Adjust code format to follow CS2103T format requirements

Commit:
add user guide

add a comprehensive guide to using Duke

Commit:
add more testing

add Junit testing for sorting deadlines and events in either chronological or lexicographical manner

GarethOng committed:
Merge pull request #6 from GarethOng/branch-A-MoreTesting

add more testing

Commit:
add Ui.png

Commit:
display Ui.png

GarethOng committed:
Update README.md

GarethOng committed:
Update README.md

Commit:
add colour

Commit:
add personality

change background colour and font

GarethOng committed:
Merge pull request #8 from GarethOng/branch-A-Personality

Branch a personality

Commit:
update Ui.png

GarethOng committed:
Update README.md

Commit:
add user guide

Commit:
Merge branch 'master' of github.com:GarethOng/ip

Commit:
add more error handling

GarethOng committed:
Merge pull request #9 from GarethOng/branch-A-MoreErrorHandling

add more error handling
